bad spec if an exclusive flag is deprecated

After updating to sf v2.85.7 (current RC, will be promoted to stable next week), `carapace-spec` panics when parsing the new `sf.yml`:

```
panic: Failed to find flag "async" and mark it as being in a mutually exclusive flag group

goroutine 1 [running]:
github.com/spf13/cobra.(*Command).MarkFlagsMutuallyExclusive(0x140004d3508, {0x14000594000, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/flag_groups.go:70 +0x20c
...
```

I think it's related to this PR: https://github.com/salesforcecli/plugin-data/pull/1222, `exclusiveFlags` has `async` but `flags` doesn't.

```yaml
- description: Bulk export records from an org into a file using a SOQL query.
    Uses Bulk API 2.0.
  name: bulk
  flags:
    --json: Format output as json.
    --flags-dir=: Import flag values from a directory.
    -o, --target-org=: Username or alias of the target org. Not required if the
      `target-org` configuration variable is already set.
    --api-version=: Override the api version used for api requests made by this
      command
    -w, --wait=: Time to wait for the command to finish, in minutes.
    -q, --query=: SOQL query to execute.
    --query-file=: File that contains the SOQL query.
    --all-rows: Include records that have been soft-deleted due to a merge or
      delete. By default, deleted records are not returned.
    --output-file=: File where records are written.
    -r, --result-format=: Format to write the results.
    --column-delimiter=: Column delimiter to be used when writing CSV output. Default is
      COMMA.
    --line-ending=: Line ending to be used when writing CSV output. Default value on
      Windows is is `CRLF`; on macOS and Linux it's `LR`.
    --help: Show help for command
  exclusiveflags:
    - - async
      - wait
    - - query
      - query-file
  completion:
    flag:
      target-org:
        - $(node
          /Users/cdominguez/.local/share/sf/node_modules/@cristiand391/sf-plugin-fzf-cmp/lib/shell-completion.js
          'sf -o ')
      result-format:
        - csv
        - json
      column-delimiter:
        - BACKQUOTE
        - CARET
        - COMMA
        - PIPE
        - SEMICOLON
        - TAB
      line-ending:
        - LF
        - CRLF
  commands: []
```

--- a/src/commands/carapace-gen.ts
+++ b/src/commands/carapace-gen.ts
@@ -126,7 +126,7 @@ https://github.com/carapace-sh/carapace-spec`
             if (node.flags[flagName].exclusive) {
               const exclusives = node.flags[flagName].exclusive.filter(flag => {
                 // @ts-ignore
-                return node.flags[flag] && !node.flags[flag].hidden
+                return node.flags[flag] && !node.flags[flag].hidden && !node.flags[flag].deprecated
               })
 
               if (exclusives.length === 0) continue

